core/src/main/java/com/graphhopper/routing/weighting/FlexModelWeighting.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting;
+
+import com.graphhopper.routing.profiles.BooleanEncodedValue;
+import com.graphhopper.routing.profiles.EncodedValueFactory;
+import com.graphhopper.routing.profiles.EncodedValueLookup;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.FlexModel;
+import com.graphhopper.routing.util.HintsMap;
+import com.graphhopper.routing.weighting.flex.AverageSpeedFlexConfig;
+import com.graphhopper.routing.weighting.flex.DelayFlexConfig;
+import com.graphhopper.routing.weighting.flex.PriorityFlexConfig;
+import com.graphhopper.util.EdgeIteratorState;
+
+import static com.graphhopper.util.EdgeIterator.NO_EDGE;
+
+/**
+ * Every EncodedValue like road_environment can influence one or more aspects of this Weighting: the delay, the
+ * speed_factor, the average_speed and the priority. The formula is basically:
+ * <pre>
+ * if no access to edge then return infinity
+ * seconds = toSeconds(distance / speed) + delayInSeconds
+ * distanceInfluence = distance * distanceFactor
+ * weight = (seconds + distanceInfluence) / priority;
+ * return weight
+ * </pre>
+ */
+public class FlexModelWeighting implements Weighting {
```

Comment:
Maybe better call this `FlexWeighting` or even `FlexibleWeighting` ? The fact that it has a `FlexModel` to represent its configuration seems like an implementation detail to me.

Reply:
There is also a bit of a name clash between flexible weighting and flexible mode (somehow related but not necessarily and should not be confused)

Reply:
Hmmh, indeed. Or maybe Custom(izable)Weighting or Config(urable)Weighting?

> The fact that it has a FlexModel to represent its configuration seems like an implementation detail to me.

Yes. On the other hand they are tightly coupled, but surely there could be a different Weighting with the same effect.

Reply:
I think I like `CustomWeighting`